### PR TITLE
[Fleet] Remove host and port for Fleet Server

### PR DESCRIPTION
**Pull request by nchaulet**
## Description 

Remove the host and port field from the Fleet server integration as it seems those values are only configurable during Fleet server installation.

Related to [154674](https://github.com/elastic/kibana/pull/154674)

cc @jen-huang @michel-laterman 

Should we restrict that package to newest version of Kibana (8.7?)

## Tests

I tested manually:
* that a fleet server is available when not providing any port or host
* that you can customize fleet server port with `--fleet-server-port`

### CODE REVIEW

**Review by kpollich — APPROVED** [review]

File: packages/fleet_server/agent/input/agent.yml.hbs [hunk 1]
```diff
@@ -1,10 +1,4 @@
 server:
```
Comment (jen-huang) [inline packages/fleet_server/agent/input/agent.yml.hbs:1]: if none of the optional fields below are present, is there any issue with having this empty `server` value?

**Review by jen-huang — COMMENTED** [review]
@nchaulet If these fields were never used in any version of Fleet Server, I don't think we need to restrict a higher Kibana version. However, I would like to check on what will happen if these vars are defined in preconfiguration, but is not present on the latest version of the package.

The reason I am asking is because on Cloud, our stackpack has these fields defined and they will always use the latest version of the Fleet Server package available to the version for that instance:
https://github.com/elastic/cloud-assets/blob/master/stackpack/kibana/config/kibana-7.yml
https://github.com/elastic/cloud-assets/blob/master/stackpack/kibana/config/kibana.yml

Could you check the behavior? If there is an issue, then we may need to restrict these changes to >v8.8.0.

**Comment by nchaulet** [comment]
> Could you check the behavior? If there is an issue, then we may need to restrict these changes to >v8.8.0.

@jen-huang I just tested a fleet server pre-configuration with some port and hosts and looks like we do not validate extra variables so we should be okay here.

I just did a test with fleet server 7.17 and in this case the host and port for the package policy is used, to be safe I think it will make sense to restrict that change to 8.8+

**Comment by jen-huang** [comment]
> I just did a test with fleet server 7.17 and in this case the host and port for the package policy is used, to be safe I think it will make sense to restrict that change to 8.8+

@nchaulet Thanks for catching that. Let's restrict it then. Could you also update the 8.x stackpack so that we don't add these fields for 8.8+?

**Review by jen-huang — APPROVED** [review]

**Review by michalpristas — COMMENTED** [review]
looks good

**Pull request merged**